### PR TITLE
chore: funding rates pagination support

### PR DESCRIPTION
**Pull request by shane-moore**
##Changes
- updated indexer-api to 1.0.4
- added pagination support to `fetchFundingRates`

### CODE REVIEW

File: packages/sdk-ts/src/client/indexer/grpc/IndexerGrpcDerivativesApi.ts [hunk 1]
```diff
@@ -469,8 +469,9 @@ export class IndexerGrpcDerivativesApi extends BaseConsumer {
         DerivativeSubaccountTradesListResponse,
         typeof InjectiveDerivativeExchangeRPC.SubaccountTradesList
       >(request, InjectiveDerivativeExchangeRPC.SubaccountTradesList)
-
-      return IndexerGrpcDerivativeTransformer.tradesResponseToTrades(response)
+      return IndexerGrpcDerivativeTransformer.tradesResponseToTrades(
+        response as DerivativeTradesResponse,
```
Comment (shane-moore) [inline packages/sdk-ts/src/client/indexer/grpc/IndexerGrpcDerivativesApi.ts:473]: @bangjelkoski I was getting typing error for the `response` parameter stating that it needs `response` as type `DerivativeTradesResponse`. curious what you think here. i'm also curious if this has anything to do with the dex failing earlier on `DerivativeTradesRequest and SpotTradesRequest`

File: packages/sdk-ts/src/client/indexer/grpc/IndexerGrpcSpotApi.ts [hunk 2]
```diff
@@ -288,8 +288,9 @@ export class IndexerGrpcSpotApi extends BaseConsumer {
         SpotSubaccountTradesListResponse,
         typeof InjectiveSpotExchangeRPC.SubaccountTradesList
       >(request, InjectiveSpotExchangeRPC.SubaccountTradesList)
-
-      return IndexerGrpcSpotTransformer.tradesResponseToTrades(response)
+      return IndexerGrpcSpotTransformer.tradesResponseToTrades(
+        response as SpotTradesResponse,
```
Comment (shane-moore) [inline packages/sdk-ts/src/client/indexer/grpc/IndexerGrpcSpotApi.ts:292]: @bangjelkoski ^^ same comment as above

File: packages/sdk-ts/src/client/indexer/types/derivatives.ts [hunk 3]
```diff
@@ -191,6 +192,11 @@ export interface FundingRate {
   timestamp: number
 }
 
+export interface FundingRateWithPagination {
```
Comment (bangjelkoski) [inline packages/sdk-ts/src/client/indexer/types/derivatives.ts:195]: Not the pattern we do, you don't need to specify the exact interface it returns, or if you want you can just do it as 
```ts
export const transformer = (response: Response): {
  fundingRates: FundingRate[]
  pagination: ExchangePagination
} => { /* */ }
```

File: packages/sdk-ts/src/client/indexer/grpc/IndexerGrpcSpotApi.ts (hunk 2 above)
Comment (bangjelkoski) [reply packages/sdk-ts/src/client/indexer/grpc/IndexerGrpcSpotApi.ts:292]: We should try and resolve this without typecasting (same for above. Maybe the import in the `IndexerGrpcSpotTransformer` is from the `exchange-api` instead?

**Review by bangjelkoski — CHANGES_REQUESTED** [review]

File: packages/sdk-ts/src/client/indexer/types/derivatives.ts (hunk 3 above)
Comment (shane-moore) [reply packages/sdk-ts/src/client/indexer/types/derivatives.ts]: @bangjelkoski got it, I destroyed this interface

File: packages/sdk-ts/src/client/indexer/grpc/IndexerGrpcSpotApi.ts (hunk 2 above)
Comment (shane-moore) [reply packages/sdk-ts/src/client/indexer/grpc/IndexerGrpcSpotApi.ts]: @bangjelkoski this was resolved by merging in `master` branch, which updated the method call to `IndexerGrpcSpotTransformer.subaccountTradesListResponseToTradesList(response)`, so we're all good here. also was resolved for `IndexerGrpcDerivativeTransformer`

**Review by bangjelkoski — APPROVED** [review]

**Pull request merged**